tests/ci/integration/run_openssh_integration.sh

```diff
@@ -80,24 +79,25 @@ function openssh_run_tests() {
 mkdir -p "${AWS_LC_BUILD_FOLDER}" "${AWS_LC_INSTALL_FOLDER}" "${OPENSSH_INSTALL_FOLDER}"
 
 # Get latest OpenSSH version.
-git clone https://github.com/openssh/openssh-portable.git
+git clone https://github.com/openssh/openssh-portable.git "${OPENSSH_WORKSPACE_FOLDER}"
 ls
 
-# Buld AWS-LC as a shared library
+# Build AWS-LC as a shared library
 aws_lc_build "$SRC_ROOT" "$AWS_LC_BUILD_FOLDER" "$AWS_LC_INSTALL_FOLDER" -DBUILD_TESTING=OFF -DBUILD_TOOL=OFF -DCMAKE_BUILD_TYPE=RelWithDebInfo -DBUILD_SHARED_LIBS=1
 install_aws_lc
 
-CODEBUILD_SKIPPED_TESTS="agent-subprocess forwarding multiplex channel-timeout forward-control agent-restrict connection-timeout"
+if [ "$OPENSSH_BRANCH" != "master" ]; then
+  checkout_openssh_branch "$OPENSSH_BRANCH"
+fi
```

Comment:
can you do this checkout unconditionally? if OpenSSH ever changes the name of their default branch, this might break.

Reply:
Git was complaining about a 'master' branch already existing, I couldn't find a nice option to checkout if it doesn't already exist so i went with this.

Reply:
hm, maybe you could do something like

```
if [ "$OPENSSH_BRANCH != $(git branch --show-current) ]; then
    checkout_openssh_branch "$OPENSSH_BRANCH
fi
```